src/corppa/poetry_detection/chadwyck_healey/tml_parser.py

```diff
@@ -682,13 +688,16 @@ def process_file(self, file_path: Path):
                 return None, None
 
             # parse with BeautifulSoup
-            # TODO: This is definitely not the right parser to use!
-            soup = BeautifulSoup(content, "html.parser")
+            soup = BeautifulSoup(content, "lxml")
+
             # extract metadata from the header section
             metadata = self.extract_metadata(soup)
             metadata["filename"] = file_path.name
+
+            if self.metadata_only:
+                return metadata, None
```

Comment:
ok, reasonable for the current structure. suggest adding a comment
```suggestion

            # in metadata-only mode, bail out before text extraction logic, returning None for poetry_text
            if self.metadata_only:
                return metadata, None
```